### PR TITLE
ci: Skip DB ops during install completely on cache hit

### PR DESCRIPTION
**Pull request by BYK**
Follow up to #3488

A new record: 2m 8s for installing self-hosted:
image

### CODE REVIEW

**Comment by BYK** [comment]
@aldy505 @aminvakil 👋🏻

File: install/set-up-and-migrate-database.sh [hunk 1]
```diff
@@ -1,33 +1,36 @@
 echo "${_group}Setting up / migrating database ..."
 
-# Fixes https://github.com/getsentry/self-hosted/issues/2758, where a migration fails due to indexing issue
-$dc up --wait postgres
+if [[ -z "${SKIP_DB_MIGRATIONS:-}" ]]; then
```
Comment (aldy505) [inline install/set-up-and-migrate-database.sh:3]: Should this be made visible on the `.env` file?

Comment (BYK) [reply install/set-up-and-migrate-database.sh:3]: Nope. Had to add this purely for CI for now. I think it's more dangerous/harmful to advertise it.

**Review by hubertdeng123 — APPROVED** [review]

**Review by aminvakil — COMMENTED** [review]
Sorry I've been late to the party, LGTM, thank you!

**Pull request merged**